services/awardSvc.js

```diff
@@ -4,9 +4,56 @@ function addMsgCounter(){
     return counter;
 }
 
+//var MongoClient = require('').MongoClient;
+//var url = "";
 
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
 
-/* var counter = 0
-function addMsg(){
-    counter += 1;
-}*/
\ No newline at end of file
+let DailyUserPostCounterSchema = new Schema({
+    gmeUserId: {type: Number, required: true},
+    gmeGroupId: {type: Number, required: true},
+    messageCount: {type: Number, required: true},
+    date: {type: Date, required: true},
+});
+
+var User =   mongoose.model('DailyUserPostCounter', DailyUserPostCounterSchema);
+// Export the model
+module.exports = User;
+
+/* look up mongo syntax,
+
+groupmeuserid: , date: ,
+
+*/
+
+// get user
+function getUser(){
+    return this.User;
+}
+
+//sort and rank
+var array = []
+function swap(array, i, j){
+    var temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+}
+function sSort(array){
```

Comment:
I brought lodash into this project. Google lodash. It'll probably be something in rankUser like var results = _.sortBy(array, 'sortValue')